src/sentences/mtw.rs

```diff
@@ -0,0 +1,111 @@
+use nom::{
+    character::complete::{char, one_of},
+    combinator::opt,
+    number::complete::float,
+    sequence::preceded,
+    IResult,
+};
+
+use crate::{parse::NmeaSentence, Error, SentenceType};
+
+/// MTW - Mean Temperature of Water
+///
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_mtw_mean_temperature_of_water>
+///
+/// ```text
+///        1   2 3
+///        |   | |
+/// $--MTW,x.x,C*hh<CR><LF>
+/// ```
+/// 1:  Temperature, degrees
+/// 2:  Unit of Measurement, Celsius
+/// 3:  Mandatory NMEA checksum
+#[derive(Debug, PartialEq)]
+pub struct MtwData {
+    pub temperature: Option<f32>,
+    pub temperature_unit: Option<MtwUnit>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum MtwUnit {
+    Celcius,
+}
+
+/// # Parse MTW message
+///
+/// Information from mtw:
+///
+/// NMEA 0183 standard Mean Temperature of Water.
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_mtw_mean_temperature_of_water>
+///
+/// ## Example (Ignore the line break):
+/// ```text
+/// $INMTW,17.9,C*1B
+///```
+///
+/// 1:  17.9         Temperature, degrees
+/// 2:  C            Unit of Measurement, Celsius
+/// 3:  *16          Mandatory NMEA checksum
+pub fn parse_mtw(sentence: NmeaSentence) -> Result<MtwData, Error> {
+    if sentence.message_id != SentenceType::MTW {
+        Err(Error::WrongSentenceHeader {
+            expected: SentenceType::MTW,
+            found: sentence.message_id,
+        })
+    } else {
+        Ok(do_parse_mtw(sentence.data)?.1)
+    }
+}
+
+fn do_parse_mtw(i: &str) -> IResult<&str, MtwData> {
+    let (i, temperature_value) = opt(float)(i)?;
+    let (i, unit) = opt(preceded(char(','), one_of("C")))(i)?;
+    let unit = unit.map(|ch| match ch {
+        'C' => MtwUnit::Celcius,
```

Comment:
Are there any other options apart from `Celsius`? I didn't find anything else, so maybe it's a good idea to just remove the field from the `MtwData`. The documentation can be specific for the field - only Celsius.

Reply:
I've had searched for this but it's kinda strange, it's described as a field in the documentation (and implemented as is in other libs) but this field seems to can only be celsius. Also didn't find any example with something else, so i assume we can remove it.

Reply:
Awesome!